### PR TITLE
Import babel-plugin-transform-dotall-regex

### PR DESCRIPTION
**Pull request by mathiasbynens**
Original: https://github.com/mathiasbynens/babel-plugin-transform-dotall-regex

Moving it into the official Babel repository makes it easier to maintain the transform.

### CODE REVIEW

**Comment by mathiasbynens** [comment]
PTAL @thejameskyle. Feel free to edit the patch directly — especially for small nits that’s much easier/faster for everyone.

File: packages/babel-plugin-transform-dotall-regex/test/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import path from 'path';
+import fs from 'fs';
+import assert from 'assert';
+import { transform, transformFileSync } from '@babel/core';
```
Comment (xtuc) [inline packages/babel-plugin-transform-dotall-regex/test/index.js]: `transform` is not used, we can remove it.

Comment (mathiasbynens) [reply packages/babel-plugin-transform-dotall-regex/test/index.js]: Done.

**Comment by mathiasbynens** [comment]
@xtuc Thanks! Patch updated.

**Comment by mathiasbynens** [comment]
@hzoo Thanks; I’ve changed `test/index.js`.

**Comment by nicolo-ribaudo** [comment]
@mathiasbynens `input.js` needs to be rename to `actual.js` in tests

**Comment by mathiasbynens** [comment]
@nicolo-ribaudo Thanks for pointing to that example. I’ve updated the patch.

As a side comment, I’m struggling to understand how the `actual.js` file name makes sense for the input. In tests, the term “actual” usually refers to the dynamically generated output based on the input, not to the input itself.

File: packages/babel-plugin-transform-dotall-regex/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Compile regular expressions using [the `s` (`dotAll`) flag](https://github.com/tc39/proposal-regexp-dotall-flag) to ES5 that works in today’s environments.
+
+For example, `/./s` is transpiled into `/[\0-\uFFFF]/`, and `/./su` is transpiled into `/[\0-\u{10FFFF}]/u`.
+
+[Here’s an online demo.](https://mothereff.in/regexpu#input=const+regex+%3D+/foo.bar/s%3B%0Aconsole.log%28%0A++regex.test%28%27foo%5Cnbar%27%29%0A%29%3B%0A//+%E2%86%92+true&dotAllFlag=1)
+
+## Installation
+
+```sh
+$ npm install @babel/plugin-transform-dotall-regex
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-dotall-regex/README.md]: `--save-dev`

**Comment by nicolo-ribaudo** [comment]
> As a side comment, I’m struggling to understand how the actual.js file name makes sense for the input. In tests, the term “actual” usually refers to the dynamically generated output based on the input, not to the input itself.

Yeah, you are right... I think that it is some sort of tecnichal debt that no one ever cared about fixing.

PS. Still `mv -t packages/babel-plugin-transform-dotall-regex/test/fixtures/dotall-regex packages/babel-plugin-transform-dotall-regex/test/fixtures/{simple,with-unicode-flag}` :wink:

**Comment by hzoo** [comment]
Yeah we should refactor the tests to just be input/output files or something, it's always been actual/expected and it's confusing

**Comment by existentialism** [comment]
@mathiasbynens updated the readme, and the tests

**Comment by hzoo** [comment]
Made https://github.com/babel/babel/issues/7063 @mathiasbynens

**Comment by existentialism** [comment]
After we land this, I have a small follow-up PR that will enable this in preset-env as well

**Comment by existentialism** [comment]
@mathiasbynens thanks! :P

**Comment by mathiasbynens** [comment]
> After we land this, I have a small follow-up PR that will enable this in preset-env as well

We’ll also want to start thinking about preset-es2018, which would include this transform.

**Comment by mathiasbynens** [comment]
Thanks for the help, everyone! ♥

**Comment by hzoo** [comment]
> preset-es2018

We are going to stop making yearly presets since we have preset-env now 😄

**Comment by mathiasbynens** [comment]
> We are going to stop making yearly presets since we have preset-env now 😄

That’s even better news! 🎉

**Pull request merged**